cou/steps/analyze.py

```diff
@@ -74,14 +74,20 @@ async def _populate(cls) -> list[OpenStackApplication]:
             for app, app_status in juju_status.applications.items()
         }
         # remove non-supported charms that return None on AppFactory.create
-        apps.remove(None)
-        upgradeable_apps = {app for app in apps if app and app.charm in UPGRADE_ORDER}
-        unknown_apps = apps - upgradeable_apps
-        upgradeable_apps_sorted = sorted(
-            upgradeable_apps, key=lambda app: UPGRADE_ORDER.index(app.charm)
+        apps.discard(None)
+        apps_to_upgrade_in_order = {app for app in apps if app and app.charm in UPGRADE_ORDER}
+        disordered_upgrade_apps = apps - apps_to_upgrade_in_order
```

Comment:
'disordered' is not conveying what you're intending to (apps that can be upgraded in any order). Perhaps a very simple `other_o7k_apps` would be enough.

I also suggest explaining the upgrade order in the docstring. For example:
```
Applications that must be upgraded in a specific order will be returned first, followed by applications that can be upgraded in any order.
Applications that are not supported will be ignored.
```